operator/watchers/node_taint.go

```diff
@@ -99,26 +110,33 @@ func checkAndMarkNode(c kubernetes.Interface, nodeGetter slimNodeGetter, nodeNam
 		if (options.RemoveNodeTaint && hasAgentNotReadyTaint(node)) ||
 			(options.SetCiliumIsUpCondition && !HasCiliumIsUpCondition(node)) {
 			logger.Info("Cilium pod running for node; marking accordingly", logfields.NodeName, node.GetName())
-
-			markNode(c, nodeGetter, node.GetName(), options, true, logger)
+			return markNode(ctx, c, nodeGetter, node.GetName(), options, true, logger)
 		}
 	} else if scheduled { // Taint nodes where the pod is scheduled but not running
 		if options.SetNodeTaint && !hasAgentNotReadyTaint(node) {
 			logger.Info("Cilium pod scheduled but not running for node; setting taint", logfields.NodeName, node.GetName())
-			markNode(c, nodeGetter, node.GetName(), options, false, logger)
+			return markNode(ctx, c, nodeGetter, node.GetName(), options, false, logger)
 		}
 	}
-	return true
+	return nil
 }
 
-// ciliumPodsWatcher starts up a pod watcher to handle pod events.
-func ciliumPodsWatcher(wg *sync.WaitGroup, clientset k8sClient.Clientset, stopCh <-chan struct{}, logger *slog.Logger) {
-	ciliumQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "cilium-pod-queue")
+func ciliumPodHandler(obj interface{}, queue workqueue.TypedRateLimitingInterface[string], logger *slog.Logger) {
+	if pod := informer.CastInformerEvent[slim_corev1.Pod](obj); pod != nil {
+		nodeName := pod.Spec.NodeName
+		// Pod might not yet be scheduled to a node
+		if nodeName != "" {
+			queue.Add(nodeName)
+		}
+	}
+}
 
+// ciliumPodsWatcher starts up a pod watcher to handle pod events.
+func ciliumPodsWatcher(wg *sync.WaitGroup, slimClient slimclientset.Interface, queue workqueue.TypedRateLimitingInterface[string], stopCh <-chan struct{}, logger *slog.Logger) {
```

Comment:
This makes me wonder: is there already a shared pod informer we can use? No sense in two separate informers (yes, I know this is a label-scoped informer, but still)

Reply:
I think there is this pod informer as well: https://github.com/cilium/cilium/blob/8dc4449edb12f8bffb4797deb7b513a92a2a61e9/operator/watchers/pod.go#L41

Ideally, we would convert both into Resource[T], but I wanted to keep this change minimal and simplify the structure a bit.